stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -695,13 +695,11 @@ internal func _copyCollectionToContiguousArray<
 
   var p = result.firstElementAddress
   var i = source.startIndex
-  for _ in 0..<count {
-    // FIXME(performance): use _copyContents(initializing:).
-    p.initialize(to: source[i])
-    source.formIndex(after: &i)
-    p += 1
-  }
-  _expectEnd(of: source, is: i)
+  var buffer = UnsafeMutableBufferPointer(start: p, count: count)
+  source._copyContents(initializing: buffer)
```

Comment:
you should probably not discard the results, and sanity check them

Reply:
I'm not sure I understand what you mean here. Could you clarify?

Reply:
`_copyContents` returns an iterator/index pair giving remaining elements and the point written up to. In theory, those should be no-remaining-elements/endIndex I think, but you could add a sanity check for that being true in debug builds.

Reply:
Ah, that makes sense. That's a great idea